cpp/arcticdb/pipeline/index_segment_reader.cpp

```diff
@@ -20,8 +20,13 @@ using namespace arcticdb::proto::descriptors;
 namespace arcticdb::pipelines::index {
 
 IndexSegmentReader get_index_reader(const AtomKey &prev_index, const std::shared_ptr<Store> &store) {
-    auto [key, seg] = store->read_sync(prev_index);
-    return index::IndexSegmentReader{std::move(seg)};
+    return async_get_index_reader(prev_index, store).get();
```

Comment:
I think we should leave this implementation using `read_sync`, so that the scheduling overhead can be avoided if necessary